### ISSUE TITLE
[Novice Question] How to display results of training in Tensorboard etc.

### ISSUE THREAD

**Issue by george-nehma** [danijar/dreamerv3#182]
I am new to RL training and was able to get Dreamerv3 running and training and can see the outputs on the terminal, however I was wondering if there is a way to display the results from training like the Tensorboard command in Dremerv2? 

I ran the training script from the README:
`python dreamerv3/main.py \
  --logdir ~/logdir/{timestamp} \
  --configs crafter \
  --run.train_ratio 32`

but would like to know if there is something like `tensorboard --logdir ~/logdir` that is in the Dreamerv2 repo?

**Comment by danijar** [comment]
We're using [Scope](https://github.com/danijar/scope) for logging by default, it has fewer dependencies and is faster. The command for viewing metrics in your browser is in the readme:

```sh
python -m scope.viewer --basedir ... --port 8000
```

You can use `--logger.outputs jsonl,tensorboard` if you want to use TensorBoard, but it is slower, has dependencies on TensorFlow and protobuf, and doesn't support MP4 (only GIF, which can get very large).

**Comment by george-nehma** [comment]
I was able to begin training on the loconav.py environment with the same code from the README but replacing crafter with loconav and I believe it is training correctly as I get this in the terminal:

> --------------------[Agent Step 558_096]--------------------
Metrics filtered by: 'score|length|fps|ratio|train/loss/|train/rand/'
train/loss/con 0.02 / train/loss/dyn 7.71 / train/loss/policy -3.3e-3 / train/loss/rep 7.71 / train/loss/repval 0 / train/loss/rew 0 / train/loss/value 0 / train/loss/walker_appendages_pos 9.6e-4 / train/loss/walker_bodies_pos 1.9e-3 / train/loss/walker_bodies_quats 5e-3 / train/loss/walker_body_height 8.9e-5 / train/loss/walker_egocentric_camera 2.71 / train/loss/walker_end_effectors_pos 9.2e-4 / train/loss/walker_joints_pos 5.5e-3 / train/loss/walker_joints_vel 0.12 / train/loss/walker_sensors_accelerometer 0.38 / train/loss/walker_sensors_gyro 0.03 / train/loss/walker_sensors_touch 0.08 / train/loss/walker_sensors_velocimeter 0.01 / train/loss/walker_world_zaxis 3.4e-4 / replay/replay_ratio 32.5 / fps/policy 44.22 / fps/train 1415.19
Writing metrics: /home/george/logdir/dreamer/20250411T170652/metrics.jsonl
Writing metrics: /home/george/logdir/dreamer/20250411T170652/scores.jsonl`

I tried running the script you sent above in another terminal whilst the training was running but the Scope is just blank.

I thought maybe the basedir needed to point to the logdir/dreamer/20250411T170652 from the training but still nothing shows up on the Scope.

Is there something am I doing wrong or misunderstanding?

**Comment by danijar** [comment]
I think you need one more parent directory for Scope to find the results.

Try moving your logdir to e.g. `~/logdir/dreamer/run1` and pointing scope to `~/logdir`. You'll have to click the refresh icon.

You can also check if there's a `scope` folder inside `run1` to double check that your training run was writing scope metrics.

**Comment by george-nehma** [comment]
Yep, changing the argument `--basedir ~/logdir` did the trick! Thank you again.
